postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -481,3 +507,21 @@ def _explain_pg_stat_activity(self, rows):
                     1,
                     tags=self._tags + ["error:collect-plan-for-statement-crash"],
                 )
+
+    def _get_query_max_text_size(self):
+        with self._get_db(self._config.dbname).cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
```

Comment:
Should we handle read errors here and proceed assuming the default track_activity_query_size of 1024 (or maybe just skip the "is statement truncated logic") if we fail to read it? 

It could be that for some reason the agent is restricted from reading `pg_settings` in which case we should be able to still proceed gracefully and continue to collect samples.

Reply:
I think this one requires some thought. I'm hesitant to default to a value that might be wrong 🤔

Reply:
I added a warning log + error tagged counter increment when we fail to load the pg_setting. And instead of using a default value that might be wrong, I had `_is_statement_truncated` return `False`. However, I think this also potentially sends the wrong information (in case the query _was_ truncated but we had no way to tell). Should I make the `statement_truncated` a string (enum style with tree value: true, false, unknown)?

Reply:
> Should I make the `statement_truncated` a string (enum style with tree value: true, false, unknown)?

That's a good idea. It would be helpful to have the "unknown" state as well so we can use it it as a signal that the agent has failed to read the config value.

Reply:
Done (using an enum instead of a bool).